src/components/Alert/index.js

```diff
@@ -0,0 +1,114 @@
+import React from 'react';
+import classNames from 'classnames';
+import CSSTransitionGroup from 'react-transition-group/CSSTransitionGroup';
+
+const { Component, PropTypes } = React;
+const FirstChild = ({ children }) => (
+  React.Children.toArray(children)[0] || null
+);
+
+class Alert extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = { isOpen: true };
+
+    this.close = this.close.bind(this);
+  }
+
+  close() {
+    this.setState({ isOpen: false });
+    if (this.props.onDismiss) {
+      this.props.onDismiss();
+    }
+  }
+
+  render() {
+    const {
+      children,
+      className,
+      dismissable,
+      emphasized,
+      id,
+      onDismiss,
+      transitionAppearTimeout,
+      transitionEnterTimeout,
+      transitionLeaveTimeout,
+      type,
+      title,
+      ...attributes
+    } = this.props;
+
+    const classes = classNames(
+      'resin-alert',
+      className,
+      { emphasized },
+      { dismissable },
+      `resin-alert-${type}`,
+    );
+
+    const alert = (
+      <div
+        {...attributes}
+        role="alert"
+        id={id}
+        className={classes}
+      >
+        { onDismiss || dismissable ?
```

Comment:
You can simplify this to onDismiss || dismissable && <button..

https://facebook.github.io/react/docs/conditional-rendering.html#inline-if-with-logical--operator

Saves having to do the `: null`